### DISCUSSION TITLE
hydra.job.chdir from python script

### DISCUSSION THREAD

**Discussion by dagap** [facebookresearch/hydra#2384] [Q&A] [answered]
I am currently creating a hydra config programmatically as follows:

```
class HydraConfigManager:
    def __init__(self, config_dir: str, config_name: str, job_name: str, overrides: List[str]) -> None:
        with initialize(version_base="1.2", config_path=config_dir, job_name=job_name):
            self.cfg = compose(
                config_name=config_name,
                overrides=overrides,
                return_hydra_config=True,
            )
```

Now, the `overrides` contains the option `hydra.job.chdir=True`. However, this does not seem to be working as expected or not taken into account. When I wrap it through `@hydra.main`, all the output gets written to the `outputs` directory but calling it as before outputs everything in the current working directory.

**Comment by Jasha10** [comment, marked as answer]
Indeed, `hydra.job.chdir` does not have any effect when used with the `compose` API; it is only relevant for the `hydra.main` API.
The motivation for this difference is that `compose` is supposed to be a [pure function](https://en.wikipedia.org/wiki/Pure_function), while `hydra.main` is not a pure function and is has side-effects (such as performing a `chdir` operation).
When using the `compose` API, my recommendation would be to manually change the working directory in your user code (e.g. by calling [`os.chdir`](https://docs.python.org/3/library/os.html#os.chdir)).
